pkg/migration-tool/planner/block.go

```diff
@@ -0,0 +1,105 @@
+package planner
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/schollz/progressbar/v3"
+	"github.com/timescale/promscale/pkg/migration-tool/utils"
+)
+
+// createBlock creates a new block and returns reference to the block for faster write and read operations.
+func (p *Plan) createBlock(mint, maxt int64) (reference *Block, err error) {
```

Comment:
if this is a method of Plan it should be in planner.go. In general I'd like to see a better separation between blocks and plans